src/structures/Message.js

```diff
@@ -571,8 +571,9 @@ class Message extends Base {
    */
   get pinnable() {
     return (
-      this.type === 'DEFAULT' &&
-      (!this.guild || this.channel.permissionsFor(this.client.user).has(Permissions.FLAGS.MANAGE_MESSAGES, false))
+      ['DEFAULT', 'REPLY', 'APPLICATION_COMMAND', 'THREAD_STARTER_MESSAGE'].includes(this.type) &&
+      (!this.guild || this.channel.permissionsFor(this.client.user).has(Permissions.FLAGS.MANAGE_MESSAGES, false)) &&
```

Comment:
`permissionsFor(...)` can return null

Reply:
How so?

Reply:
if the client user's guildmember (`Guild#me`) is uncached then it will return null

Reply:
I believe that can never be uncached unless the user disables the cache which is expected. I’m not sure why this isn’t in the warning in ClientOptions#makeCache though. Is it really worth adding optional chaining?